### ISSUE TITLE
Page image Error - mit neue Contao 4.11

### ISSUE THREAD

**Issue by Cleanerrr** [terminal42/contao-pageimage#46]
`[2021-02-23T12:51:49.316887+01:00] request.INFO: Matched route "tl_page.2.root". {"route":"tl_page.2.root","route_parameters":{"_token_check":true,"_controller":"Contao\\FrontendIndex::renderPage","_scope":"frontend","_locale":"de","_format":"html","parameters":"","pageModel":{"Contao\\PageModel":[]},"_route":"tl_page.2.root","_route_object":{"Symfony\\Component\\Routing\\Route":[]}},"request_uri":"http://mywebsite.com/","method":"GET"} []
[2021-02-23T12:51:49.344184+01:00] security.INFO: Populated the TokenStorage with an anonymous Token. [] []
[2021-02-23T12:51:49.409114+01:00] request.CRITICAL: Uncaught PHP Exception LogicException: ""contao.fragment._contao.frontend_module.pageimage" has no container set, did you forget to define it as a service subscriber?" at /www/htdocs/mywebsitepath/vendor/symfony/framework-bundle/Controller/ControllerResolver.php line 39 {"exception":"[object] (LogicException(code: 0): \"contao.fragment._contao.frontend_module.pageimage\" has no container set, did you forget to define it as a service subscriber? at /www/htdocs/mywebsitepath/vendor/symfony/framework-bundle/Controller/ControllerResolver.php:39)"} []`

**Comment by xprojects-de** [comment]
Jip, habe den gleichen Fehler:

[2021-03-16T07:21:15.937312+01:00] request.CRITICAL: Uncaught PHP Exception LogicException: ""contao.fragment._contao.frontend_module.pageimage" has no container set, did you forget to define it as a service subscriber?" at .../contao4/vendor/symfony/framework-bundle/Controller/ControllerResolver.php line 39 {"exception":"[object] (LogicException(code: 0): \"contao.fragment._contao.frontend_module.pageimage\" has no container set, did you forget to define it as a service subscriber? at .../contao4/vendor/symfony/framework-bundle/Controller/ControllerResolver.php:39)"} []

**Comment by Cleanerrr** [comment]
Hallo,
ich habe gleiche Probleme mit andere Erweiterungen, wie mmenu, Rocksolid Slider,...
Das Problem kann neue Contao 4.11. sein? Vieleicht...#
Wenn ich letzte 4.10.* version instaliert, habe ich kein Probleme mit Erweiterungen. 
Ich kann nicht neue Contao 4.11.* verwenden wegen diese Problem.
Haben Sie änliche Erfahrungen/Probleme ?

**Comment by ab2211** [comment]
Ich habe den Fehler mit Contao 4.11 wenn ich die Erweiterung "Bild pro Seite" aktiviere.

**Comment by fritzmg** [comment]
I haven't really looked into this problem yet.

**Comment by dp-designpilot** [comment]
Ich habe auch das gleiche Problem und würde die Erweiterung dringend benötigen (kannst Du das fixen Andy??)

**Comment by xprojects-de** [comment]
Hi,

Maybe, if NO autowire is set, the container must be given to the service via "call"? Isn't that a special case for the AbstractController? Actually thought that you don't have to do that anymore since the 4-version.
But i can be wrong...!

calls: ['setContainer', ...Containerinterface]

**Comment by fritzmg** [comment]
@dp-designpilot as a work around you could let `symfony/framework-bundle` be downgraded to `4.4.*`, either by a `require` or `conflict` for `"symfony/framework-bundle": ">=5"`.

**Comment by dp-designpilot** [comment]
> @dp-designpilot as a workaround you could let `symfony/framework-bundle` be downgraded to `4.4.*`, either by a `require` or `conflict` for `"symfony/framework-bundle": ">=5"`.

Hallo Fritz
Wie muss ich das genau machen, ich hab sowas noch nie gemacht ???

**Comment by dp-designpilot** [comment]
Hallo Fritz
Ich habe das gemacht was Du gesagt hast jetzt ist es aber so, das ich wahrscheinlich Fehler
in den beiden Erweiterungen habe (wahrscheinlich rührt das wegen dem folgenden Eintrag in der composer.json) 

"symfony/framework-bundle": ">=5"

https://github.com/hofff/contao-tawk-to/issues/1
https://github.com/hofff/contao-facebook-pixel/issues/3

**Comment by richardhj** [comment]
We must set the container on the controller manually since Symfony 5.

I updated framework-bundle to Symfony 5 and NONE of my Page Controllers, Content Elements, Modules, or other Fragment Controllers were working.

First, I found the issue symfony/symfony#36495 that suggests that Controller services or their aliases must be public. I also updated the RegisterPagesPass / RegisterFragmentsPass to make those definitions public, but that's not working.

Other tickets (e.g. symfony/symfony#33755 and following) indicate that you must manually set the container to the service. This shouldn't be done via a compiler pass (see ticket).

After all, this remains the best fix:

```yml
services:
  _instanceof:
    Symfony\Bundle\FrameworkBundle\Controller\AbstractController:
      calls:
        - [ setContainer, [ '@Psr\Container\ContainerInterface' ] ]
```
